caddyhttp/proxy/upstream_test.go

```diff
@@ -153,6 +153,8 @@ func TestAllowedPaths(t *testing.T) {
 		{"/proxy//static", false},
 		{"/proxy//static//download", false},
 		{"/proxy//download", false},
+		{"/proxy/trailingslash", true},
+		{"/proxy/trailingslash/", false},
```

Comment:
Can you please add a test case for a path consisting of only "/"?

Reply:
In order to test this, I'd need to add `/` to the list of ignored sub paths as this code only executes on ignored sub paths.

That would make every possible path ignored by the proxy, causing every `AllowedPath` test to return `false`.

While I _would_ also argue "who would ever `except /` from a proxy?", I acknowledge that someone might do it sometime and it needs to not break horrifically...

Should I set up another testing function with only `/` ignored? Called `TestAllowedPathTrailingSlashOnly` or something? 🤔

Reply:
Those are good concerns. I will revise this to use `strings.HasSuffix` and to check both that the string originally had a trailing slash _and_ no longer has one after cleaning.

I am still not sure what the best way to test those cases would be, though.